src/ethereum/arrow_glacier/vm/instructions/block.py

```diff
@@ -38,13 +38,18 @@ def block_hash(evm: Evm) -> None:
 
     # OPERATION
     max_block_number = block_number + Uint(256)
-    if evm.env.number <= block_number or evm.env.number > max_block_number:
+    if (
```

Comment:
```suggestion
    block_number = evm.message.block_env.number
    if (
```

Introducing a temporary will make the next couple of lines more readable.

Reply:
Updated. `block_number` is already assigned to the value that we pop from the stack. So, I just went with `current_block_number`